binaries/llm-cli/src/main.rs

```diff
@@ -283,26 +297,40 @@ fn interactive(
         };
         sp.clear();
 
-        session.infer::<Infallible>(
-            model.as_ref(),
-            &mut rng,
-            &llm::InferenceRequest {
-                prompt: "".into(),
-                parameters: &parameters,
-                play_back_previous_tokens: false,
-                maximum_token_count: generate.num_predict,
-            },
-            &mut Default::default(),
-            |r| match r {
-                InferenceResponse::PromptToken(t) | InferenceResponse::InferredToken(t) => {
-                    print!("{t}");
-                    std::io::stdout().flush().unwrap();
+        if chat_mode {
+            session.infer::<Infallible>(
+                model.as_ref(),
+                &mut rng,
+                &llm::InferenceRequest {
+                    prompt: "".into(),
+                    parameters: &parameters,
+                    play_back_previous_tokens: false,
+                    maximum_token_count: generate.num_predict,
+                },
+                &mut Default::default(),
+                conversation_inference_callback(stop_sequence.clone(), &mut buf, print_token),
+            )
+        } else {
+            session.infer::<Infallible>(
+                model.as_ref(),
+                &mut rng,
+                &llm::InferenceRequest {
+                    prompt: "".into(),
```

Comment:
Are the prompts meant to be empty here?

Reply:
ooh great find! i don't suspect so. this was in the previous impl, let me try with `prompt: prompt.as_str().into(),` and test out

Reply:
no that's not it. I think it needs to be `message_prompt_template` or `initial_prompt_template` depending on the call number. But with that kind of branching logic, it might be time to more fully separate these code paths

Reply:
Oh, disregard, I'm a silly billy - I didn't see the `feed_prompt` above the `infer`. (I think the `feed_prompt` can be removed and replaced with `infer` entirely, but might need to test that.) This is fine as-is.